File: chapter_1/1.1/1.1.4/Square.java
// Jordi Becerril Enriquez
// November 08, 2024
// java square example


public class Square {
  public static void main(String[] args) {
    
    double result = square(5);
    System.out.println(result);

    double res = sumOfSquares(3, 4);
    System.out.println(res);

    double fres = f(5);
    System.out.println(fres);
  }

  public static double square(double x) {
    return x * x;
  }

  public static double sumOfSquares(double x, double y) {
    double a = square(x);
    double b = square(y);
    return a + b;
  }

  public static double f(double a) {
    return sumOfSquares((a + 1), (a * 2));
  }
}

File: chapter_1/1.1/1.1.4/square.py
# Jordi Becerril Enriquez
# November 08, 2024
# python square example


def square(x):
    return x * x

print(square(5)) # 25

def sum_of_squares(x, y):
    a = square(x)
    b = square(y)
    return a + b

print(sum_of_squares(3, 4)) #25

# Further Procedures:
def f(a):
    return sum_of_squares((a + 1), (a * 2))

print(f(5)) #136

